### ISSUE TITLE
removeEventSource differentiate between sources with same URL

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#3209]
if you call `removeEventSource` and there are multiple sources with the same URL, [as jdhorner points out](https://github.com/fullcalendar/fullcalendar/issues/2384#issuecomment-162059246), all get removed.

http://jsbin.com/hijupufaxa/1/edit?js,output

**Comment by jdhorner** [comment]
Hey Adam,
My team looked at this, and some thought maybe it shouldn't even be a bug, and it works as they thought it should. In that, thinking in terms of simple RESTful endpoints as sources, because they have the same URL, they _are_ the same source. (Even if passed different parameters, like in my previous example)
```javascript
myEventSources = {
  staff202: {
    url: "/StoreAdmin/Resources/GetEvents",
    type: "GET",
    data: { id: 202 },
    traditional: true,
    color: "green"
  },
  staff102: {
    url: "/StoreAdmin/Resources/GetEvents",
    type: "GET",
    data: { id: 102 },
    traditional: true,
    color: "green"
  }
};
```
We decided the more "correct" way to do it would be to include the IDs as part of the URLs, like so: 
```javascript
myEventSources = {
  staff202: {
    url: "/StoreAdmin/Resources/GetEvents/202",
    type: "GET",
    traditional: true,
    color: "green"
  },
  staff102: {
    url: "/StoreAdmin/Resources/GetEvents/102",
    type: "GET",
    traditional: true,
    color: "green"
  }
};
```
and of course, it worked as expected, as the URLs now no longer match. 

I'm not sure how that affects all this, but I thought you should know. I found it interesting that, depending on whom I asked, they expected different behavior. "Of course they should both be deleted, they're the same source!" vs. "They're defined separately, they should be treated as such!"

**Comment by arshaw** [comment]
i think if you pass in a single string argument to `removeEventSource`, it should remove all sources with that URL, but if you pass in the original object for the event source, it should remove only that specific event source with a `===` comparison to that object.

**Comment by arshaw** [comment]
[Code complete](https://github.com/fullcalendar/fullcalendar/commits/v2.8.0-dev) and [docs written](https://github.com/fullcalendar/fullcalendar/commits/docs-2.0) but not yet released. Will update this thread when released.

**Comment by arshaw** [comment]
Released in [v2.8.0](https://github.com/fullcalendar/fullcalendar/releases/tag/v2.8.0).

same behavior if you pass in a URL to `removeEventSource`, but more precise behavior if you pass in the original object.
